do-not-commit/useless-eqeq.py

```diff
@@ -0,0 +1,27 @@
+# ruleid:useless-eqeq
+x == x
```

Comment:
This is always True: `x == x` or `x != x`. If testing for floating point NaN, use `math.isnan(x)`,
or `cmath.isnan(x)` if the number is complex. ([.useless-eqeq.useless-eqeq](https://semgrep.dev/r/.useless-eqeq.useless-eqeq))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.useless-eqeq.useless-eqeq) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>